x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinatorTests.java

```diff
@@ -2350,6 +2353,79 @@ public void testFollowerIndexIsCreatedInExecuteAutoFollow() {
         assertThat(autoFollowResults.v2().contains(indexName), equalTo(true));
     }
 
+    public void testRemovesStatsOnRemovingAutoFollowPattern() {
+        // given auto-follow pattern added
+        var autoFollowCoordinator = createAutoFollowCoordinator();
+        autoFollowCoordinator.updateAutoFollowers(
+            createClusterStateWith(
+                createAutoFollowPattern("pattern1", "logs-*"),
+                createAutoFollowPattern("pattern2", "logs-*"),
+                createAutoFollowPattern("pattern3", "logs-*")
+            )
+        );
+
+        // and stats are produced
+        autoFollowCoordinator.updateStats(
+            List.of(
+                new AutoFollowCoordinator.AutoFollowResult("pattern1", new RuntimeException("ClusterStateFetchException")),
+                new AutoFollowCoordinator.AutoFollowResult(
+                    "pattern1",
+                    List.of(Tuple.tuple(new Index("index1", UUIDs.base64UUID()), new RuntimeException("AutoFollowExecutionException")))
+                )
+            )
+        );
+        // and applied
+        assertThat(autoFollowCoordinator.getStats().getRecentAutoFollowErrors(), hasKey("pattern1:index1"));
```

Comment:
I was surprised that the second AutoFollowResult cleared the error from the first one. 
Not sure if this is intended behavior.